packages/web-components/src/components/footer/footer.scss

```diff
@@ -85,28 +102,47 @@
   }
 }
 
+:host(#{$dds-prefix}-footer) {
+  ::slotted(#{$dds-prefix}-language-selector-mobile) {
+    margin-top: $spacing-09;
+
+    @include carbon--breakpoint('md') {
+      margin-top: 0;
+      @include carbon--make-col(4, 8);
+      @include carbon--make-col-offset(2, 8);
+    }
+
+    @include carbon--breakpoint('lg') {
+      display: none;
+    }
+  }
+
+  ::slotted(#{$dds-prefix}-language-selector-desktop) {
+    @include carbon--breakpoint('lg') {
+      @include carbon--make-col(4, 16);
+      @include carbon--make-col-offset(10, 16);
+    }
+  }
+}
+
 :host(#{$dds-prefix}-footer[size='short']) {
   ::slotted(#{$dds-prefix}-language-selector-desktop) {
     margin-top: 0;
 
     @include carbon--breakpoint('lg') {
-      flex: 0 0 25%;
-      margin-left: 62.5%;
-    }
-
-    @include carbon--breakpoint('max') {
-      flex: 0 0 18.75%;
-      margin-left: 68.75%;
+      @include carbon--make-col(4, 16);
+      @include carbon--make-col-offset(10, 16);
```

Comment:
I think we can concat this to the previous block and reduce some repeated styles, i.e.
```scss
:host(#{$dds-prefix}-footer) {
  ...

  &[size='short'] {
    // no more `carbon--breapoint('lg')` needed for this or `-language-selector-mobile`,
    // or `carbon--breakpoint('md')`, etc
  }
}
```